account_credit_control/models/credit_control_communication.py

```diff
@@ -85,43 +90,35 @@ def _compute_total(self):
             communication.total_invoiced = communication._get_total()
             communication.total_due = communication._get_total_due()
 
-    @api.model_create_multi
-    @api.returns('self', lambda value: value.id)
-    def create(self, vals_list):
-        for vals in vals_list:
-            if vals.get('partner_id'):
-                # the computed field does not work in TransientModel,
-                # just set a value on creation
-                partner_id = vals['partner_id']
-                vals['contact_address'] = \
-                    self._get_contact_address(partner_id).id
-        return super(CreditControlCommunication, self).create(vals_list)
+    @api.model
+    def _onchange_partner_id(self):
+        """Update address when partner changes."""
+        for one in self:
+            partners = one.env["res.partner"].search([
+                ("id", "child_of", one.partner_id.id),
+            ])
+            if one.contact_address_id in partners:
+                # Contact is already child of partner
+                return
+            address_ids = one.partner_id.address_get(adr_pref=['invoice'])
+            one.contact_address_id = address_ids["invoice"]
 
     @api.multi
     def get_email(self):
         """ Return a valid email for customer """
         self.ensure_one()
-        contact = self.contact_address
+        contact = self.contact_address_id
         email = contact.email
         if not email and contact.commercial_partner_id.email:
             email = contact.commercial_partner_id.email
         return email
 
     @api.multi
     @api.returns('res.partner')
-    def get_contact_address(self):
```

Comment:
You shouldn't rename this method, as you get a record, not an ID

Reply:
Oops, this rename came from a bulk search/replace. But I'm seeing that this method does nothing, so I'm just gonna add the needed migration code and remove this method.